### DISCUSSION TITLE
Integration with ts-patch

### DISCUSSION THREAD

**Discussion by iserdmi** [justkey007/tsc-alias#221] [Ideas]
Hello! Sorry, don't you want convert this solution to ts-patch plugin? https://github.com/nonara/ts-patch as optional solution in external repo?
It can help to avoid huge commands for watch and build. Jut want to ask you. Thanks!

**Comment by kevinswarner** [comment]
I would also love to see this as a ts-patch plugin. My use case is that we have a complex watch setup in a monorepo where a consuming application watches for changes of it's dependent libraries, and for the dependent library, we need to not only compile with tsc, but also transform the path aliases. Because it is a two step process, and the non-transformed .js files exist in the outDir for a short time, the consuming app sees the updated .js file and throws an error because it cannot resolve the path. Sometimes, it corrects itself once the transform is done, but not always. It would be nice to have the compile and transform as one atomic step.

Currently, I am attempting a solution that builds and transforms in a temp directory, and then uses rsync to sync from that directory to the dist/ directory, which is what the consuming applications look to. But, I still have not gotten around to the watching part yet, so I am not sure if I can get it working.
